netadapter/server/grpcserver/grpc_connection.go

```diff
@@ -0,0 +1,93 @@
+package grpcserver
+
+import (
+	"net"
+	"sync/atomic"
+
+	"github.com/pkg/errors"
+
+	"github.com/kaspanet/kaspad/netadapter/server"
+	"github.com/kaspanet/kaspad/netadapter/server/grpcserver/protowire"
+	"github.com/kaspanet/kaspad/wire"
+	"google.golang.org/grpc"
+)
+
+type gRPCConnection struct {
+	server                *gRPCServer
+	address               net.Addr
+	sendChan              chan *protowire.KaspadMessage
+	receiveChan           chan *protowire.KaspadMessage
+	errChan               chan error
+	clientConn            grpc.ClientConn
+	onDisconnectedHandler server.OnDisconnectedHandler
+
+	isConnected int32
+}
+
+func newConnection(server *gRPCServer, address net.Addr) *gRPCConnection {
+	connection := &gRPCConnection{
+		server:      server,
+		address:     address,
+		sendChan:    make(chan *protowire.KaspadMessage),
+		receiveChan: make(chan *protowire.KaspadMessage),
+		errChan:     make(chan error),
+		isConnected: 1,
+	}
+
+	server.addConnection(connection)
+
+	return connection
+}
+
+func (c *gRPCConnection) IsConnected() bool {
+	return atomic.LoadInt32(&c.isConnected) == 1
```

Comment:
Please use != 0 instead. I've encountered naughty programmers using -1 for true as well.